### PR TITLE
feat: add optional opt out and opt in

### LINKED ISSUE #5: Opt Out & Opt In
Enable support for both "opting in" and "opting out" model on the repository-level. 
- The main issue I was inspired by right now is that I would like to be able to quickly create new repos and issues in old repos under @ubiquity without having to manually add each repo here every time. 
- I would like for us to be able to add approved partners, like us (@ubiquity) and automatically import all of their public repositories. 

`opt.json`

```json
{
    "out": [
        "ubiquity/devpool-directory",
        "ubiquity/.github"
    ],
    "in": [
        "alpha-labs-global/blockalizer",
        "uniswap/interface"
    ]
}
```

- Most partners will use `opt.in` to opt in specific repositories. 
- Special partners, including @Ubiquity will automatically sync all of their issues to the directory, and will be using `opt.out` to exclude repositories.

---

### Notice

- It is implied that we automatically import all of the repositories from the `ubiquity` organization. 
- "`ubiquity/`" can be parsed within `opt.out` as an organization name.

### PR DESCRIPTION
**Pull request by gitcoindev**
Enable support for both "opting in" and "opting out" model on the repository-level.

opt.json configuration file may contain multiple repostiories and orgs, e.g.

{
    "out": [
        "ubiquity/devpool-directory",
        "ubiquity/.github"
    ],
    "in": [
        "uniswap/interface"
    ]
}

Resolves: https://github.com/ubiquity/devpool-directory-bounties/issues/5

### CODE REVIEW

**Review by rndquu — APPROVED** [review]

**Comment by gitcoindev** [comment]
Hi @rndquu thank you for reviewing and the approval! Seems that the pull request needs to be manually merged, no automatic merging was enabled.

**Comment by rndquu** [comment]
> Hi @rndquu thank you for reviewing and the approval! Seems that the pull request needs to be manually merged, no automatic merging was enabled.

Let's wait for 1 more approval

File: index.ts
Comment (0x4007) [inline index.ts:255]: I think its unusal to use switch case in this context but given that the codebase is small and simple this should not be an issue.

**Review by 0x4007 — APPROVED** [review]

**Comment by 0x4007** [comment]
We have an event handler in the bot that automatically opens a pull request to add a specific repo upon bot installation to the directory. I think this might need adjustments due to this now. @gitcoindev what do you think?

**Pull request merged**
